src/smhi/smhi.py

```diff
@@ -78,11 +79,50 @@ def find_stations_from_gps(
 
         user_position = (latitude, longitude)
         self.get_stations(parameter)
-        self.d = []
-
-        all_stations = self.client.stations.data
-        self.d = [
-            s
-            for s in all_stations
-            if distance.distance(user_position, (s["latitude"], s["longitude"])) <= dist
-        ]
+        self.nearby_stations = []
+        all_stations = self.client.stations.stations
+        if dist == 0:
+            stations = [
+                (s['id'], s['name'], distance.distance(
+                    user_position, (s["latitude"], s["longitude"])).km)
+                for s in all_stations
+            ]
+            self.nearby_stations = min(stations, key=lambda x: x[2])
+        else:
+            self.nearby_stations = [
+                (s['id'], s['name'], distance.distance(
+                    user_position, (s["latitude"], s["longitude"])).km)
+                for s in all_stations
+                if distance.distance(user_position, (s["latitude"], s["longitude"])) <= dist
+            ]
+            self.nearby_stations = sorted(self.nearby_stations, key=lambda x: x[2])
+
+    def find_stations_by_city(
+        self, parameter: int, city: str, dist: float = 0
+    ) -> None:
+        """Find stations for parameter from city name.
+
+        Args:
+            parameter: station parameter
+            dist: distance from city
+            city: name of city
+        """
+
+        geolocator = Nominatim(user_agent='ifk-smhi')
+        loc = geolocator.geocode(city)
+        self.find_stations_from_gps(parameter=parameter, dist=dist,
+                                    latitude=loc.latitude, longitude=loc.longitude)
+
+    def get_data(
+        self, parameter: int, station: int, period: str = 'corrected-archive'
+    ) -> None:
```

Comment:
This method returns a tuple of something (is it str and some sort of list)? Strange that mypy doesn't fail here.